src/hnsw/build.c

```diff
@@ -166,6 +221,11 @@ static void InitBuildState(HnswBuildState *buildstate, Relation heap, Relation i
 
     buildstate->dimensions = GetHnswIndexDimensions(index);
 
+    // If a dimension wasn't specified try to infer it
+    // this is unecessary when the actual size is 3 but in practice this shouldn't be too common
+    if(columnType == REAL_ARRAY || columnType == INT_ARRAY)
+        if(buildstate->dimensions == HNSW_DEFAULT_DIMS)
```

Comment:
I think we should set `HNSW_DEFAULT_DIMS` to -1, NULL or something like that, as now if we pass `(dims=3)` on index creation and insert array with 4 dimensions it is going to overwrite the `dims` option.